doc/source/conf.py

```diff
@@ -110,6 +110,16 @@
 # The full version, including alpha/beta/rc tags.
 release = strictversion
 
+# Substitutions defined for all pages
+rst_prolog = f"""
+.. |silx_installer_btn| replace::
+   .. link-button:: https://github.com/silx-kit/silx/releases/download/v{version}/silx-{version}-windows-installer-x86_64.exe
+      :classes: btn-success
+      :text: Download Windows installer
+
+.. |silx_archive| replace:: :download:`silx ZIP archive <https://github.com/silx-kit/silx/releases/download/v{version}/silx-{version}-windows-application.zip>`
+"""
```

Comment:
That's the way sphinx/rst offers to tune documentation content from the config file: it is not possible to do substitutions inside directives/roles (e.g. use `|version|` inside a link)...